Cauli/Cauli.swift

```diff
@@ -35,8 +35,8 @@ public class Cauli {
     /// The Storage used by this instance to store all Records.
     public let storage: Storage
     internal let florets: [Floret]
-    private var enabledFlores: [Floret] {
-        return florets.filter { $0.enabled }
+    private var enabledFlorets: [InterceptableFloret] {
+        return florets.lazy.compactMap { $0 as? InterceptableFloret }.filter { $0.enabled }
```

Comment:
Afaik, the `lazy` here is useless, since you filter upon the lazy collection and return an array anyways.